src/com/google/enterprise/gsafeed/GsafeedHelper.java

```diff
@@ -40,6 +41,10 @@ public void setErrorHandler(ErrorHandler errorHandler) {
     this.errorHandler = errorHandler;
   }
 
+  public void setValidationEventHandler(ValidationEventHandler eventHandler) {
```

Comment:
Is this really part of the API? What's the use case for this (beyond the tests)?

Reply:
Same as the reason for exposing the error handler: I don't want to assume that the simplest possible implementation is all that's ever called for. Someone could want a different sort of logging or something, or have a use case where they don't care if the dates are wrong. It's also possible to switch to use javax.xml.bind.util.ValidationEventCollector to accumulate events instead of failing immediately.